hetio/hetnet.py

```diff
@@ -274,9 +274,17 @@ def add_edge(self, edge_id):
             inverse.inverted = True
             self.n_inverts += 1
 
-    def extract_metapaths(self, source_kind, target_kind, max_length):
-        source = self.node_dict[source_kind]
-        target = self.node_dict[target_kind]
+    def extract_metapaths(self, source, target=None, max_length=4):
```

Comment:
Any reason we pick default `max_length=4` here but no default in `extract_all_metapaths`?

Reply:
you cannot have args without defaults come after args with defaults. I didn't want to set max_length, but had to if I wanted to set a default for target. I'm open to setting in both places if you think that's best or choosing a different default like 0 to make sure people always set it.

Reply:
That makes sense. Of the numbers to pick, I think 4 is  a good default.